src/data/dataset.py

```diff
@@ -101,3 +103,59 @@ def collate_fn(self, samples: list[Dialog], return_all_candidates: bool = True):
         b_candidates = b_candidates.view(len(samples), -1, b_candidates.size(1))
 
         return b_contexts, b_candidates.squeeze(1)
+
+
+class CommonsenseConversationDataset(Dataset):
+    def __init__(self, path, tokenizer_name, max_context_len, max_target_len=None, have_candidates=True):
+        self.dataset = []
+
+        self.context_tokenizer = AutoTokenizer.from_pretrained(tokenizer_name, truncation_side="left")
+        self.candidate_tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+        self.max_context_len = max_context_len
+        self.max_target_len = max_target_len or max_context_len
+
+        self.have_candidates = have_candidates
+        self.vocab_size = self.context_tokenizer.vocab_size
+
+        logger.info(f"Loading dataset from '{path}'")
+
+        with open(path, "r") as f:
+            lines_list = list(f)
+
+        for line in lines_list:
+            result = json.loads(line)
+            dialog = {"context": [result["src"]], "candidates": [result["trg"]]}
+            self.dataset.append(Dialog(**dialog))
+
+        logger.info(f"Loaded {len(self.dataset)} samples")
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def __getitem__(self, idx: int) -> Dialog:
+        return self.dataset[idx]
+
+    def collate_fn(self, samples: list[Dialog]):
```

Comment:
It appears that you are copying the matching function from the ConvAI dataset, but this is not necessary. We don't have any candidates for the Commonsense dataset, so things are much simpler

```python
def __getitem__(self, idx: int) -> tuple[str, str]:
        return self.dataset[idx]```

def collate_fn(self, samples: list[tuple[str, str]]):
    str_contexts, str_candidates = zip(*samples)

    # [batch size, context seq len]
    b_contexts = self.context_tokenizer(
            str_contexts,
            max_length=self.max_context_len,
            padding=True,
            truncation=True,
            return_tensors="pt",
            add_special_tokens=False,
        ).input_ids

        # Tokenizer truncates on the left, but for candidates we want to truncate on the right
        b_candidates = self.candidate_tokenizer(
            str_candidates, padding="max_length", return_tensors="pt", add_special_tokens=False
        ).input_ids

        return b_contexts, b_candidates
```